packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -726,8 +728,8 @@ export class MarketOperationUtils {
                         optimizerOpts,
                     );
                 }
-            } else {
-                // A firm quote is being requested. Ensure that `intentOnFilling` is enabled.
+            } else if (!rfqt.isIndicative && isFirmPriceAwareEnabled) {
```

Comment:
should there be an 'else if...' if priceAwareEnabled is false for both firm and indicative?

Reply:
good question! if firm and indicative quotes are both disabled, none of that if/else block is triggered. In this scenario, the `optimizedOrders` variable is never replaced and proceeds with what was previously set before. This is expected.

Does that make sense? or am I missing smth?

Reply:
We have routes for

(rfqt.isIndicative && isIndicativePriceAwareEnabled)
and
(!rfqt.isIndicative && isFirmPriceAwareEnabled)

but was wondering if we needed routes for:

(rfqt.isIndicative && !isIndicativePriceAwareEnabled)
and
(!rfqt.isIndicative && !isFirmPriceAwareEnabled)

Or are we only using RFQT if it's price aware?

Reply:
oh, that's a really good question. So, this pathway that starts [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R683) is only accessed when price-aware RFQ is enabled. When Price-aware RFQ is disabled, we still do RFQ but at a different section of the code: [here](https://github.com/0xProject/protocol/pull/30/files#diff-24bb4e154d234a6991fb599588e16823703a26243956e07cacd9ea7722de7ba4L703) for firm quotes and [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R218) for indicative quotes

Reply:
oh, that's a really good question. So, this pathway that starts [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R683) is only accessed when price-aware RFQ is enabled. When Price-aware RFQ is disabled, we still do RFQ but at a different section of the code: [here](https://github.com/0xProject/protocol/pull/30/files#diff-24bb4e154d234a6991fb599588e16823703a26243956e07cacd9ea7722de7ba4L703) for firm quotes and [here](https://github.com/0xProject/protocol/pull/30/files#diff-863cb8194bd0f4e5a026d8ba07c4a5256dacd6e342a427d6953fa759b28f8239R218) for indicative quotes